### ISSUE TITLE
Make remote network and storage names configurable in settings

### ISSUE THREAD

**Issue by idroz** [bravetools/bravetools#147]
Current remote settings:

```json
{
    "name": "bravetools",
    "url": "https://192.168.64.51:8443",
    "protocol": "lxd",
    "public": false,
    "profile": "profile_name"
}
```

Suggested remote settings:

```json
{
    "name": "bravetools",
    "url": "https://192.168.64.51:8443",
    "protocol": "lxd",
    "public": false,
    "storage":  "storage_name",
    "network":  "profile_namebr0",
    "profile": "profile_name"
}
```

This would allow to have fine-grained control over remote deployments and won't force user to have identical local and remote configurations.

**Comment by Szubie** [comment]
I think this makes sense. It will certainly make remotes easier to use once they've been set up, despite the number of settings to specify when creating them.

I would view profile/network/storage specified in the remote as the overridable defaults for the remote - at the moment the Profile is also specifiable in the Bravefile itself or via CLI at deploy time. It may make sense to do this for storage/network too.

Note that we currently rely on storage devices available to the LXD profile provided - supporting attaching specific storage devices to specific containers would be more flexible.

**Comment by idroz** [comment]
>at the moment the Profile is also specifiable in the "Service" section of the Bravefile itself or via CLI at deploy time. It may make sense to do this for storage/network too.

This makes sense - I didn't realise that the Service section of the Bravefile supported that. I think it is the right place to have that configuration. Remotes could have multiple profiles/networks and registered remotes in ~/.bravetools should only have the minimal configuration that let's us establish the first handshake with a remote. Therefore, a Bravefile is a logical place to customise deployment options

**Comment by idroz** [comment]
THsi is addressed in #153
